megamek/src/megamek/client/Client.java

```diff
@@ -161,6 +129,15 @@
 
     public Map<String, Client> bots = new TreeMap<String, Client>(StringUtil.stringComparator());
 
+    //get the image data
+    private MechTileset mechTileset = new MechTileset(Configuration.unitImagesDir());
```

Comment:
Why not use the TilesetManager class instead of basically rolling your own implementation of it? It's present in the BoardView class and contains pretty much all of the information you could need.

Reply:
Initially I was having trouble getting an image to show, so after I got it working initially I just kept building around it. Now that I have a better understand of how it works, I'll definitely go back and do it that way and address the other changes as well.